### PR TITLE
Refs #28643 -- Added Repeat database function.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28643

### CODE REVIEW

**Comment by felixxm** [comment]
Failures on MySQL are not related with this PR, it returns wrong results also for other functions, e.g. `LPad('name', Length('alias'))`. I'll investigate :male_detective: .

**Comment by felixxm** [comment]
It turns out that `mysqlclient` returns bytes instead of strings in some cases (see similar [issue](https://github.com/PyMySQL/mysqlclient-python/issues/145)), therefore if I use `TextField` (with the explicit converter `convert_textfield_value`) everything works as expected.

**Comment by timgraham** [comment]
Looks good, pending the MySQL issue.

File: django/db/backends/sqlite3/base.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ def get_new_connection(self, conn_params):
         conn.create_function("django_format_dtdelta", 3, _sqlite_format_dtdelta)
         conn.create_function("django_power", 2, _sqlite_power)
         conn.create_function('LPAD', 3, _sqlite_lpad)
+        conn.create_function('REPEAT', 2, _sqlite_repeat)
```
Comment (pope1ni) [inline django/db/backends/sqlite3/base.py]: I'd make this a lambda or use `operator.mul`.

**Comment by felixxm** [comment]
I added the separate ticket [29251](https://code.djangoproject.com/ticket/29251) for the MySQL issue.

File: tests/db_functions/test_repeat.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            (Repeat('name', 2), 'JohnJohn'),
+            (Repeat('name', Length('alias'), output_field=CharField()), 'JohnJohnJohn'),
+            (Repeat(Value('x'), 3, output_field=CharField()), 'xxx'),
+        )
+        for function, repeated_text in tests:
+            with self.subTest(function=function):
+                authors = Author.objects.annotate(repeated_text=function)
+                self.assertQuerysetEqual(authors, [repeated_text], lambda a: a.repeated_text, ordered=False)
+
+    def test_repeat_negative_number(self):
```
Comment (timgraham) [inline tests/db_functions/test_repeat.py]: chop "_repeat"

File: tests/db_functions/test_repeat.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from django.db.models import CharField, Value
+from django.db.models.functions import Length, Repeat
+from django.test import TestCase
+
+from .models import Author
+
+
+class RepeatTests(TestCase):
+    def test_repeat(self):
```
Comment (timgraham) [inline tests/db_functions/test_repeat.py]: test_basic

**Comment by felixxm** [comment]
@timgraham Thanks for the review.

**Pull request merged**